tagger/api.py

```diff
@@ -78,20 +78,37 @@ def endpoint_interrogate(self, req: models.TaggerInterrogateRequest):
         if req.image is None:
             raise HTTPException(404, 'Image not found')
 
+        if isinstance(req.model, list):
+            if any(i not in utils.interrogators.keys() for i in req.model):
+                raise HTTPException(404, 'Model {i} not found')
+
         if req.model not in utils.interrogators.keys():
             raise HTTPException(404, 'Model not found')
+            req.model = [req.model]
 
         image = decode_base64_to_image(req.image)
-        interrogator = utils.interrogators[req.model]
-
-        with self.queue_lock:
-            QData.tags.clear()
-            QData.ratings.clear()
-            QData.in_db.clear()
-            QData.for_tags_file.clear()
-            data = ('', '', '') + interrogator.interrogate(image)
+        QData.tags.clear()
+        QData.ratings.clear()
+        QData.in_db.clear()
+        QData.for_tags_file.clear()
+
+        # allow overriding of default values
+        if req.threshold:
+            QData.threshold = req.threshold
+        if req.tag_frac_threshold:
+            QData.tag_frac_threshold = req.tag_frac_threshold
+        if req.count_threshold:
+            QData.count_threshold = req.count_threshold
+
+        for model in req.model:
+            interrogator = utils.interrogators[model]
+            with self.queue_lock:
+                data = ('', '', '') + interrogator.interrogate(image)
             QData.apply_filters(data)
-            output = QData.finalize(1)
+            if req.auto_unload:
+                interrogator.unload()
+
+        output = QData.finalize(1)
```

Comment:
output[3] contains a string in case of an error. Should this also be included in the TaggerInterrogateResponse?